app.js

```diff
@@ -282,11 +301,76 @@ router.post(
     }
 );
 
+const sendSecurityCode = async ({securityCode, phoneNumber, email}) => {
+
+    if (phoneNumber) {
+        await sendSms({
+            text: `Your NEAR Wallet security code is: ${securityCode}`,
+            to: phoneNumber
+        });
+    } else if (email) {
+        await sendMail({
+            to: email,
+            subject: `Your NEAR Wallet security code is: ${securityCode}`,
+            text: `Use this code to confirm your email address: ${securityCode}`
+        });
+    }
+};
+
+router.post('/account/initializeRecoveryMethod',
+    checkAccountOwnership,
+    async ctx => {
+        const { accountId, email, phoneNumber, test } = ctx.request.body;
+        const [account] = await models.Account.findOrCreate({ where: { accountId } });
+        const where = {
+            kind: phoneNumber ? 'phone' : 'email',
+            detail: phoneNumber || email
+        };
+
+        let [recoveryMethod] = await account.getRecoveryMethods({ where });
+
+        if (!recoveryMethod) {
+            recoveryMethod = await account.createRecoveryMethod(where);
+        }
+
+        const securityCode = password.randomPassword({ length: SECURITY_CODE_DIGITS, characters: password.digits });
+        await recoveryMethod.update({ securityCode });
+        await sendSecurityCode({ securityCode, phoneNumber, email });
+
+        if (test) {
+            return ctx.body = securityCode;
+        }
+
+        ctx.body = await recoveryMethodsFor(account);
+    }
+);
+
+router.post('/account/validateSecurityCode',
+    checkAccountOwnership,
+    async ctx => {
+        const { accountId, phoneNumber, email, securityCode } = ctx.request.body;
+
+        const [account] = await models.Account.findOrCreate({ where: { accountId } });
```

Comment:
hmm, shouldn't account already exist by the time we need to validate security code? Looks like shouldn't be using `findOrCreate`